main.go

```diff
@@ -961,14 +990,17 @@ func main() {
 				// Assumes that if hook channels are not nil, they will have at
 				// least one value before getting closed
 				exitCode := 0 // is 0 if all hooks succeed, else is 1
-				if exechookRunner != nil {
-					if err := exechookRunner.WaitForCompletion(); err != nil {
-						exitCode = 1
+				// This will not be needed if async == false, because the Send func for the hookRunners will wait
+				if *flHooksAsync {
+					if exechookRunner != nil {
+						if err := exechookRunner.WaitForCompletion(); err != nil {
```

Comment:
Looking at it more, this is a bit tricky - you might get a result on the channel that indicates that the hook failed, but it will be retried (async).  For one-time that's fine, but is it OK in non-one-time mode?  If you ask for synchronous hooks, and the hook fails -- do we think that's "good enough" or do we wait for a success?  What if success never comes?